design-book/src/design-constraints.md

```diff
@@ -0,0 +1,58 @@
+# Design Constraints
+
+To help ensure that we're building an editor that's both useful to end users and feasible to maintain,
+we've developed a set of design constraints and goals. Please try to follow them when making and reviewing changes!
+As we're developing, initial prototypes may bend or break the rules: just be mindful of how we can move towards the final desirable end goal.
+
+Below each bullet point, motivation is given.
+
+## Must
+
+- The editor must target the latest stable Bevy release.
+  - This eases project management by separating the work across two repositories, and ensures that we aggressively test each Bevy release's migration path.
+- The Bevy editor must be a Bevy application.
+  - This is essential both for dogfooding and to ease development: Bevy contributors all know how to write Bevy!
+- The Bevy editor must run on Windows, Mac and Linux.
+  - These desktop platforms are the key targets, as they are used by professionals for in-depth creative work. Web is a stretch goal, mobile and console need substantially more design.
+
+## Should
+
+- Reusable code (camera controllers, inspectors, UI widgets...) should be cleanly separated into dedicated, documented libraries.
+  - This keeps our repository clean, and allows users to steal our work to use in their own projects and tooling.
+- Widgets, styles, colors and interaction paradigms should be consistent across the editor.
+  - This avoids duplication, makes changes easier and helps give the editor a cohesive look.
+- The editor should be distributed as an executable, either directly from the Bevy website or via Steam.
+  - This makes it easier for non-technical users to install.
+- The editor should be configurable, and easily adapted to both different genres and specific games.
+  - Every game needs a different set of features. We need to allow teams to expose or extend what they care about, and hide what they don't.
+- Industry-standard conventions for controls, UX and so on should be followed unless there's a good reason not to.
```

Comment:
Although I agree that UX is all about familiarity, it doesn't mean that the industry standards were well thought out. Many of this standards were developed without UX and by programmers, which created standards that are definitely not easy to onboard. I would suggest here a well thought UX process and usability testing for controles, widgets and components

Reply:
The option to swap out to different source-editor hotkeys and controls would be good to think about early. For example, do we want the camera to behave like it does in Blender, or like Unity's flycam? Which set of hotkeys should we use for common functionality like Undo, Copy/Paste, etc.?

Reply:
Please, please don't create replacements for Control/Command C,V,Z,Y without need. Alternative custom bindings are fine, but don't alinenate people with 30 years muscle memory.